test/unit/merge.js

```diff
@@ -1215,72 +1216,103 @@ describe('Merge', function() {
+        'local',
+        _.cloneDeep(file2),
+        _.cloneDeep(file)
+      )
+      const was = await this.pouch.db.get(file2._id)
+      const dst = builders
+        .metadir(src)
+        .path('DST')
+        .noRev()
+        .build()
       const doc = await builders
-        .metafile()
+        .metafile(was)
         .path('DST/FILE2')
         .build()
 
       const sideEffects = await mergeSideEffects(this, () =>
-        this.merge.moveFileAsync(this.side, _.cloneDeep(doc), _.cloneDeep(was))
+        this.merge.moveFolderAsync('local', _.cloneDeep(dst), _.cloneDeep(src))
       )
 
       const movedSrc = _.defaults(
+        {
+          moveTo: dst._id,
+          _deleted: true
+        },
+        src
+      )
+      const movedFile = _.defaults(
         {
           moveTo: doc._id,
+          childMove: true,
           _deleted: true
         },
         was
       )
       should(sideEffects).deepEqual({
         savedDocs: [
-          // FIXME: missing deleted SRC document due to fake directory move
-          {
-            _id: 'DST',
-            path: 'DST',
-            docType: 'folder',
-            sides: { rev: 1, [this.side]: 1 },
-            tags: [],
-            updated_at: doc.updated_at // XXX: It might fail if it takes longer to run moveFileAsync
-          },
           _.omit(movedSrc, ['_rev']),
           _.defaults(
             {
-              sides: increasedSides(was.sides, this.side, 1),
+              sides: increasedSides(src.sides, this.side, 1),
               moveFrom: movedSrc
             },
-            doc
+            _.omit(dst, ['_rev'])
+          ),
+          _.omit(movedFile, ['_rev']),
+          _.defaults(
+            {
+              sides: increasedSides(was.sides, this.side, 1),
+              moveFrom: movedFile
+            },
+            _.omit(doc, ['_rev'])
           )
         ],
         resolvedConflicts: []
       })
     })
 
     it('does not identify the local move of a file following an unsynced child move as an addition', async function() {
-      const orig = await builders
-        .metafile()
-        .path('SRC/FILE')
-        .sides({ local: 2, remote: 2 })
+      const src = await builders
+        .metadir()
+        .path('SRC')
+        .upToDate()
         .remoteId(dbBuilders.id())
         .create()
-      const was = await builders
+      await builders
         .metafile()
-        .path('DST/FILE')
-        .moveFrom(orig)
-        .sides({ local: 1 })
+        .path('SRC/FILE')
+        .upToDate()
+        .remoteId(dbBuilders.id())
         .create()
+      const dst = builders
+        .metadir(src)
+        .path('DST')
+        .noRev()
+        .build()
+      await this.merge.moveFolderAsync(
+        'local',
+        _.cloneDeep(dst),
+        _.cloneDeep(src)
+      )
+      const was = await this.pouch.db.get(
+        builders
+          .metafile()
+          .path('DST/FILE')
+          .build()._id
```

Comment:
`metadata.id('DST/FILE')`?
Ideally we should rename `Pouch#byPath()` to something less confusing, e.g. `#byParentPath()`, then reuse the `Pouch#byPath()` name for this.

Reply:
The issue with `metadata.id()` is that it does not normalize paths. So in order to make this test pass, we either need to require the `path` module and rename every occurrence of the `path` variable to avoid collision or use the builder instead which already normalizes paths.

I don't really understand the link with `Pouch#byPath`.